python/ray/autoscaler/_private/aws/config.py

```diff
@@ -489,29 +501,70 @@ def _configure_subnet(config):
         ]
         if not subnets:
             cli_logger.abort(
-                "No usable subnets matching availability zone {} found.\n"
-                "Choose a different availability zone or try "
-                "manually creating an instance in your specified region "
-                "to populate the list of subnets and trying this again.",
-                config["provider"]["availability_zone"],
+                f"No usable subnets matching availability zone {azs} found "
+                f"for node type {node_type_key}.\nChoose a different "
+                "availability zone or try manually creating an instance in "
+                "your specified region to populate the list of subnets and "
+                "trying this again."
+            )
+        elif _are_user_subnets_pruned(subnets):
+            cli_logger.abort(
+                f"MISMATCH between specified subnets and Availability Zones! "
+                "The following Availability Zones were specified in the "
+                f"`provider section`: {azs}.\n The following subnets for node "
+                f"type `{node_type_key}` have no matching availability zone: "
+                f"{_get_pruned_subnets(subnets)}. "
             )
 
     # Use subnets in only one VPC, so that _configure_security_groups only
     # needs to create a security group in this one VPC. Otherwise, we'd need
     # to set up security groups in all of the user's VPCs and set up networking
     # rules to allow traffic between these groups.
     # See https://github.com/ray-project/ray/pull/14868.
-    subnet_ids = [s.subnet_id for s in subnets if s.vpc_id == subnets[0].vpc_id]
+    return [s.subnet_id for s in subnets if s.vpc_id == subnets[0].vpc_id]
```

Comment:
```suggestion
    subnets = [s.subnet_id for s in subnets if s.vpc_id == subnets[0].vpc_id]
    if _are_user_subnets_pruned(subnets):
        subnet_vpcs = {s.subnet_id: s.vpc_id for s in user_specified_subnets}
        cli_logger.abort(
            "Subnets specified in more than one VPC! Please ensure that all subnets "
            "share the same VPC and retry your request. Subnet VPCs: {}", subnet_vpcs
        )
    return subnets
```
Any subnets pruned here will otherwise raise an error of the following form when we try to use them to create an EC2 instance in `node_provider.py` later:
```
 create_instances: Attempt failed with An error occurred (InvalidParameter) when calling the RunInstances operation: Security group sg-000000 and subnet subnet-111111 belong to different networks., retrying.
 ```